tests/first-facets-distribution.tests.ts

```diff
@@ -0,0 +1,148 @@
+  beforeAll(async () => {
+    const deleteTask = await meilisearchClient.deleteIndex('movies')
+    await meilisearchClient.waitForTask(deleteTask.taskUid)
+    await meilisearchClient
+      .index('movies')
+      .updateFilterableAttributes(['genres'])
+    const documentsTask = await meilisearchClient
+      .index('movies')
+      .addDocuments(dataset)
+    await meilisearchClient.index('movies').waitForTask(documentsTask.taskUid)
+  })
+
+  test('creation of facet distribution without facets', async () => {
+    const searchClient = instantMeiliSearch(HOST, API_KEY)
+    const response = await searchClient.search([
+      {
+        indexName: 'movies',
+        params: {
+          facets: [],
+          query: '',
+        },
+      },
+    ])
+    expect(response.results[0].facets).toEqual({})
+  })
+
+  test('creation of facet distribution without facets and with keepZeroFacets to true', async () => {
+    const searchClient = instantMeiliSearch(HOST, API_KEY, {
+      keepZeroFacets: false,
+    })
+    const response = await searchClient.search([
+      {
+        indexName: 'movies',
+        params: {
+          facets: [],
+          query: '',
+        },
+      },
+    ])
+    expect(response.results[0].facets).toEqual({})
+  })
+
+  test('creation of facet distribution with facets', async () => {
+    const searchClient = instantMeiliSearch(HOST, API_KEY)
+    const response = await searchClient.search([
+      {
+        indexName: 'movies',
+        params: {
+          facets: ['genres'],
+          query: '',
+        },
+      },
+    ])
+    expect(response.results[0].facets).toEqual({
+      genres: {
+        Action: 3,
+        Adventure: 1,
+        Animation: 1,
+        Comedy: 2,
+        Crime: 4,
+        Drama: 1,
+        'Science Fiction': 2,
+        Thriller: 1,
+      },
+    })
+  })
+
+  test('creation of facet distribution with facets and keepZeroFacets to true', async () => {
+    const searchClient = instantMeiliSearch(HOST, API_KEY, {
+      keepZeroFacets: true,
+    })
+    const response = await searchClient.search([
+      {
+        indexName: 'movies',
+        params: {
+          facets: ['genres'],
+          query: '',
+        },
+      },
+    ])
+    expect(response.results[0].facets).toEqual({
+      genres: {
+        Action: 3,
+        Adventure: 1,
+        Animation: 1,
+        Comedy: 2,
+        Crime: 4,
+        Drama: 1,
+        'Science Fiction': 2,
+        Thriller: 1,
+      },
+    })
+
+    const response2 = await searchClient.search([
+      {
+        indexName: 'movies',
+        params: {
+          facets: ['genres'],
+          query: 'no results',
+        },
+      },
+    ])
+
+    expect(response2.results[0].facets).toEqual({
+      genres: {
+        Action: 0,
+        Adventure: 0,
+        Animation: 0,
+        Comedy: 0,
+        Crime: 0,
+        Drama: 0,
+        'Science Fiction': 0,
+        Thriller: 0,
+      },
+    })
+  })
+
```

Comment:
I think you can remove this, because you have the same setup/exercise in the test below.

Reply:
There is actually a difference for me! 
The first tests: 
- Makes a search with no query
- Should cache the default distribution correctly
- Makes a search with the same `searchClient` as before but with a query that would return 0 documents
- Checks if every facet is correctly set to 0

Second test:
- Does a search with a query that would return 0 documents
- Should cache the default distribution when none has been set before
- Checks if every facet is correctly set to 0

There is a cache system between the multiple calls on the same search client. These two tests ensure they are helping each other. 
The tests that ensure that when doing two call on my search client only triggers the default facetdistribution call only once is tested here: https://github.com/meilisearch/instant-meilisearch/pull/830#discussion_r958622286. In this test we are calling two time `searchClient.search`. During the first `search` we make two requests to meilisearch, one for the default facet distribution, one for the actual search. During the second `search` on the same `searchClient` the default facet distribution should already be cached and only the actual search request should be made, thus 3 calls.